packages/blocks/src/api/utils.js

```diff
@@ -57,11 +57,15 @@ export function isUnmodifiedDefaultBlock( block ) {
 	const newDefaultBlock = isUnmodifiedDefaultBlock.block;
 	const blockType = getBlockType( defaultBlockName );
 
-	return every(
-		blockType?.attributes,
-		( value, key ) =>
-			newDefaultBlock.attributes[ key ] === block.attributes[ key ]
```

Comment:
This is trying to compare an empty String to an empty RichTextValue. Unfortunately there's no way to intercept `===`, so converting to a String seems like the only option.